compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -520,35 +519,64 @@ private static boolean validateDotSeparatedIdentifiers(String identifiers) {
+        pkgManifestDependencies.forEach(manifestDependency -> {
+            addDependencyContent(content, manifestDependency.org().value(), manifestDependency.name().value(),
+                                 manifestDependency.version().value().toString());
+            if (manifestDependency.repository() != null) {
+                content.append("repository = \"").append(manifestDependency.repository()).append("\"\n");
+            }
+            content.append("\n");
+        });
+
+        // write dependencies from package dependency graph
+        pkgGraphDependencies.forEach(graphDependency -> {
+            PackageDescriptor descriptor = graphDependency.packageInstance().descriptor();
 
             // ignore lang libs & ballerina internal packages
-            if (!(descriptor.isLangLibPackage() || descriptor.org().value().equals(BALLERINA_INTERNAL_ORG.value))) {
-                content.append("[[dependency]]\n");
-                content.append("org = \"").append(descriptor.org().value()).append("\"\n");
-                content.append("name = \"").append(descriptor.name().value()).append("\"\n");
-                content.append("version = \"").append(descriptor.version().value()).append("\"\n");
-                dependencies.forEach(dependency1 -> {
-                    if (dependency1.org().value().equals(dependency.packageInstance().packageOrg().value())
-                            && dependency1.name().value().equals(dependency.packageInstance().packageName().value())
-                            && dependency1.repository() != null) {
-                        content.append("repository = \"").append(dependency1.repository()).append("\"\n");
-                    }
-                });
-                content.append("\n");
+            if (!descriptor.isLangLibPackage() && !graphDependency.injected()) {
+                // write dependency, if it not already exists in `Dependencies.toml`
+                if (!isPkgManifestDependency(graphDependency, pkgManifestDependencies)) {
+                    addDependencyContent(content, descriptor.org().value(), descriptor.name().value(),
+                                         descriptor.version().value().toString());
+                    content.append("\n");
+                }
             }
-        }
+        });
         return String.valueOf(content);
     }
 
+    private static boolean isPkgManifestDependency(ResolvedPackageDependency graphDependency,
+            List<PackageManifest.Dependency> pkgManifestDependencies) {
+        if (pkgManifestDependencies.isEmpty()) {
+            return false;
+        }
+
+        for (PackageManifest.Dependency manifestDependency : pkgManifestDependencies) {
+            if (manifestDependency.org().value().equals(graphDependency.packageInstance().packageOrg().value())
+                    && manifestDependency.name().value()
+                    .equals(graphDependency.packageInstance().packageName().value())) {
```

Comment:
Don't we need to trim white space in string comparison? 
`manifestDependency.name().value().trim().equals(graphDependency.packageInstance().packageName().value().trim())`

Reply:
I'm not sure why we need to do that, Were do you found issues by comparing these?

Reply:
yeah, this won't need, if we don't allow white space in packageName. :)

Reply:
```

Reply:
yeah sure :)